src/rvw.cpp

```diff
@@ -7,129 +7,225 @@
 
 // [[Rcpp::export(".create_cache")]]
 void create_cache(std::string dir="", std::string data_file="", std::string cache_file="") {
-  std::string data_str = dir + data_file;
-  std::string cache_str = dir + cache_file;
-
-  // Rcpp::Rcout << "Creating cache from: " << data_str << std::endl;
-
+    std::string data_str = dir + data_file;
+    std::string cache_str = dir + cache_file;
+    
+    // Rcpp::Rcout << "Creating cache from: " << data_str << std::endl;
+    
     // Redirect cerr
-  // std::stringstream new_buf;
-  // auto old_buf = std::cerr.rdbuf(new_buf.rdbuf());
-
-  std::string cache_init_str = "-d " + data_str + " -c --cache_file " + cache_str;
-  vw* cache_model = VW::initialize(cache_init_str);
-  VW::start_parser(*cache_model);
-
-  // Modified version of LEARNER::generic_driver that only creates cache and doesn't train
-  example* ec = nullptr;
-  while ( cache_model->early_terminate == false )
-  if ((ec = VW::get_example(cache_model->p)) != nullptr) {
-    VW::finish_example(*cache_model, ec);
-  } else {
-    break;
-  }
-  if (cache_model->early_terminate) //drain any extra examples from parser.
-    while ((ec = VW::get_example(cache_model->p)) != nullptr)
-      VW::finish_example(*cache_model, ec);
-  VW::end_parser(*cache_model);
-  VW::finish(*cache_model);
-
-  // std::cerr.rdbuf(old_buf);
-
-  // Rcpp::Rcout << "Cache file created: " << cache_str << std::endl;
-
-
+    // std::stringstream new_buf;
+    // auto old_buf = std::cerr.rdbuf(new_buf.rdbuf());
+    
+    std::string cache_init_str = "-d " + data_str + " -c --cache_file " + cache_str;
+    vw* cache_model = VW::initialize(cache_init_str);
+    VW::start_parser(*cache_model);
+    
+    // Modified version of LEARNER::generic_driver that only creates cache and doesn't train
+    example* ec = nullptr;
+    while ( cache_model->early_terminate == false )
+        if ((ec = VW::get_example(cache_model->p)) != nullptr) {
+            VW::finish_example(*cache_model, ec);
+        } else {
+            break;
+        }
+        if (cache_model->early_terminate) //drain any extra examples from parser.
+            while ((ec = VW::get_example(cache_model->p)) != nullptr)
+                VW::finish_example(*cache_model, ec);
+        VW::end_parser(*cache_model);
+        VW::finish(*cache_model);
+        
+        // std::cerr.rdbuf(old_buf);
+        
+        // Rcpp::Rcout << "Cache file created: " << cache_str << std::endl;
+        
+        
 }
 
+//'Train Vowpal Wabbit model
+//'
```

Comment:
Could you add a descriptions to the function and put a space between the `//'` and `@`? e.g.

```cpp
//' Train Vowpal Wabbit Model
//'
//' Within this function we act upon the vwmodel setup using \link{\code{vwsetup}}.... 
//' 
//' @param vwmodel Model of vw class created from \link{\code{vwsetup}} to train model.
//' ...
```